### ISSUE TITLE
Scanner and app crash on iOS 9

### ISSUE THREAD

**Issue by Coolcat13** [Anyline/anyline-ocr-cordova-module#13]
Sorry for opening a new issue - I cannot re-open #10 and not getting replies on my latest comments.

The anyline scanner opens briefly then crashes the app on iOS 9.3 - this has now been tested on a real iPad 3 (retina) and a simulated iPhone 5S - can you please let us know if you are aware of this issue and when you expect to fix it as we cannot proceed with the SDK unless it is fully supported on iOS 8+ (except for 720P devices which we can live with)

Thanks

**Comment by JonasLaux** [comment]
Hi @Coolcat13,

Unfortunately you can't simulate our SDK, cause Anyline needs a camera to work. 

The iPad 3 crash is weird though. We fully support 8+ devices with min. 1080p cameras. Can you please send some more informations like the crash log and the implementation?

**Comment by Coolcat13** [comment]
Here is the log of the error upon launching scanner in iOS 9.3 - hope this helps:

Dec  4 13:56:33 [32019]: AnylineSDK Version 3.18.0 (1353)
Dec  4 13:56:33 [32019]: CFURLSetResourcePropertyForKey failed because it was passed an URL which has no scheme

The app then crashes

Thanks

**Comment by JonasLaux** [comment]
Thank you very much. Could you please share your configuration and implementation?

**Comment by Coolcat13** [comment]
Not sure how much you want, but here you go:

- Cordova 6.5.0
- Cordova-ios 4.5.4
- Anyline cordova plugin 3.18.4
- iOS 9.3.5 on iPad 3 (real device) and iPhone 5S (simulator)

**Comment by JonasLaux** [comment]
I meant more of the stacktrace, like the log when the crash happens, just a bit more. 
Also with Configuration i meant the Anyline Configuration like [this](https://github.com/Anyline/anyline-ocr-cordova-module/blob/master/example/www/js/anyline.ocr.js#L107)
and the implementation like [this](https://github.com/Anyline/anyline-ocr-cordova-module/blob/master/example/www/js/anyline.ocr.js#L159). Or are you just using the examples app? Also, of which scanMode are we talking about?

this would help us to reproduce and investigate your error. Thanks

**Comment by Coolcat13** [comment]
Just to clarify - the configuration works perfectly on Android and iOS 10/11 - I don't have stack trace but I am using the example anyline config from the docs https://documentation.anyline.com/toc/modules/anyline_ocr/index.html

**Comment by JonasLaux** [comment]
Sorry, but without any clue what is crashing i can't help you. 
When you adding the iOS platform, Cordova is creating an XCode project for you (workspace and project). If you open this and run it e.g in XCode, there is a console which will show you a lot more log output.
Also, you can get the stacktrace after the crash with typing in "bt".
https://stackoverflow.com/questions/15946499/xcode-full-stack-trace/15946723#15946723

**Comment by Coolcat13** [comment]
Does this help?

Dec  4 15:37:42 [35031]: THREAD WARNING: ['AnylineSDK'] took '12.138184' ms. Plugin should use a background thread.
Dec  4 15:37:42 [35031]: AnylineSDK Version 3.18.0 (1353)
Dec  4 15:37:42 [35031]: CFURLSetResourcePropertyForKey failed because it was passed an URL which has no scheme
Dec  4 15:37:42 [35031]: +[NSTimer scheduledTimerWithTimeInterval:repeats:block:]: unrecognized selector sent to class 0x10a68e0c0
Dec  4 15:37:42 [35031]: *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '+[NSTimer scheduledTimerWithTimeInterval:repeats:block:]: unrecognized selector sent to class 0x10a68e0c0'
	*** First throw call stack:
	(
		0   CoreFoundation                      0x000000010a468d85 __exceptionPreprocess + 165
		1   libobjc.A.dylib                     0x0000000109ba6deb objc_exception_throw + 48
		2   CoreFoundation                      0x000000010a471c3d +[NSObject(NSObject) doesNotRecognizeSelector:] + 205
		3   CoreFoundation                      0x000000010a3b7cfa ___forwarding___ + 970
		4   CoreFoundation                      0x000000010a3b78a8 _CF_forwarding_prep_0 + 120
		5   XXXXX App                          0x00000001035f0fb9 -[ALWatermarkView startTimerForWatermarkCheck] + 137
		6   XXXXX App                          0x00000001035f0522 -[ALWatermarkView initWithCutout:screenFrame:] + 125
		7   XXXXX App                          0x00000001035ee01d -[ALAbstractScanViewPlugin evaluateWatermark:] + 718
		8   XXXXX App                          0x00000001035e8f5a -[ALAbstractScanViewPlugin startAndReturnError:] + 33
		9   XXXXX App                         0x00000001035fda57 -[ALOCRScanViewPlugin startAndReturnError:] + 51
		10  XXXXX App                          0x00000001035b6ce9 -[AnylineAbstractModuleView startScanningAndReturnError:] + 57
		11  XXXXX App                           0x0000000103259082 -[AnylineBaseScanViewController viewDidAppear:] + 210
		12  UIKit                               0x0000000107265335 -[UIViewController _setViewAppearState:isAnimating:] + 830
		13  UIKit                               0x0000000107265cb8 -[UIViewController _endAppearanceTransition:] + 262
		14  UIKit                               0x000000010723008a -[UIPresentationController transitionDidFinish:] + 827
		15  UIKit                               0x000000010740d38f -[_UICurrentContextPresentationController transitionDidFinish:] + 42
		16  UIKit                               0x00000001072337c1 __56-[UIPresentationController runTransitionForCurrentState]_block_invoke_2 + 183
		17  UIKit                               0x0000000107af24c8 -[_UIViewControllerTransitionContext completeTransition:] + 101
		18  UIKit                               0x000000010722cf77 -[UITransitionView notifyDidCompleteTransition:] + 252
		19  UIKit                               0x000000010722cc88 -[UITransitionView _didCompleteTransition:] + 1344
		20  UIKit                               0x000000010722f3f4 -[UITransitionView _transitionDidStop:finished:] + 104
		21  UIKit                               0x00000001071527ff -[UIViewAnimationState sendDelegateAnimationDidStop:finished:] + 241
		22  UIKit                               0x0000000107152bae -[UIViewAnimationState animationDidStop:finished:] + 80
		23  QuartzCore                          0x0000000106e323c8 _ZN2CA5Layer23run_animation_callbacksEPv + 308
		24  libdispatch.dylib                   0x000000010ab733d7 _dispatch_client_callout + 8
		25  libdispatch.dylib                   0x000000010ab5de5d _dispatch_main_queue_callback_4CF + 714
		26  CoreFoundation                      0x000000010a3c20f9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
		27  CoreFoundation                      0x000000010a383b99 __CFRunLoopRun + 2073
		28  CoreFoundation                      0x000000010a3830f8 CFRunLoopRunSpecific + 488
		29  GraphicsServices                    0x000000010c04cad2 GSEventRunModal + 161
		30  UIKit                               0x00000001070c4f09 UIApplicationMain + 171
		31  XXXXX                           0x000000010322fc41 main + 65
		32  libdyld.dylib                       0x000000010aba292d start + 1
		33  ???                                 0x0000000000000001 0x0 + 1
	)

**Comment by JonasLaux** [comment]
ok, we investigated the error and you'r right, we have a bug there. We will release a new SDK this week. Check back on Friday. In our 3.19 release this will be fixed

**Comment by JonasLaux** [comment]
Any news on this one?

**Comment by Coolcat13** [comment]
I can confirm the build now works using v 3.19.1, just not when we do a cordova run, but we can live with that. Thanks for your help
